examples/silentpayments.c

```diff
@@ -0,0 +1,618 @@
+    },
+    {
+        0x03, 0x0b, 0x9d, 0xd4, 0x9d, 0xf2, 0xc6, 0x85,
+        0x23, 0xbb, 0x0c, 0x72, 0xd4, 0xfb, 0x59, 0xb6,
+        0x4c, 0xe5, 0xc9, 0xa9, 0x33, 0x6d, 0x0b, 0xef,
+        0x94, 0x9e, 0xe0, 0x77, 0x5b, 0xea, 0x61, 0xef, 0x05
+    }
+};
+static unsigned char carol_scan_key[32] = {
+    0x04, 0xb2, 0xa4, 0x11, 0x63, 0x5c, 0x09, 0x77,
+    0x59, 0xaa, 0xcd, 0x0f, 0x00, 0x5a, 0x4c, 0x82,
+    0xc8, 0xc9, 0x28, 0x62, 0xc6, 0xfc, 0x28, 0x4b,
+    0x80, 0xb8, 0xef, 0xeb, 0xc2, 0x0c, 0x3d, 0x17
+};
+static unsigned char carol_address[2][33] = {
+    {
+        0x03, 0xbb, 0xc6, 0x3f, 0x12, 0x74, 0x5d, 0x3b,
+        0x9e, 0x9d, 0x24, 0xc6, 0xcd, 0x7a, 0x1e, 0xfe,
+        0xba, 0xd0, 0xa7, 0xf4, 0x69, 0x23, 0x2f, 0xbe,
+        0xcf, 0x31, 0xfb, 0xa7, 0xb4, 0xf7, 0xdd, 0xed, 0xa8
+    },
+    {
+        0x03, 0x81, 0xeb, 0x9a, 0x9a, 0x9e, 0xc7, 0x39,
+        0xd5, 0x27, 0xc1, 0x63, 0x1b, 0x31, 0xb4, 0x21,
+        0x56, 0x6f, 0x5c, 0x2a, 0x47, 0xb4, 0xab, 0x5b,
+        0x1f, 0x6a, 0x68, 0x6d, 0xfb, 0x68, 0xea, 0xb7, 0x16
+    }
+};
+
+/** Labels
+ *
+ *  The structs and call back function are implemented here as a demonstration
+ *  of how the label lookup callback is meant to query a label cache and return
+ *  the label tweak when a match is found. This is for demonstration purposes
+ *  only and not optimized. In a production usecase, it is expected that the
+ *  caller will be using a much more performant method for storing and querying
+ *  labels.
+ *
+ *  Recipients not using labels can ignore these steps and simply pass `NULL`
+ *  for the label_lookup and label_context arguments:
+ *
+ *      secp256k1_silentpayments_recipient_scan_outputs(..., NULL, NULL);
+ */
+
+struct label_cache_entry {
+    unsigned char label[33];
+    unsigned char label_tweak[32];
+};
+
+struct labels_cache {
+    size_t entries_used;
+    struct label_cache_entry entries[5];
+};
+
+const unsigned char* label_lookup(
+    const unsigned char* label33,
+    const void* cache_ptr
+) {
+    const struct labels_cache* cache = (const struct labels_cache*)cache_ptr;
+    size_t i;
+    for (i = 0; i < cache->entries_used; i++) {
+        if (my_memcmp_var(cache->entries[i].label, label33, 33) == 0) {
+            return cache->entries[i].label_tweak;
+        }
+    }
+    return NULL;
+}
+
+int main(void) {
+    enum { N_INPUTS = 2, N_OUTPUTS = 3 };
+    unsigned char randomize[32];
+    unsigned char serialized_xonly[32];
+    secp256k1_xonly_pubkey tx_inputs[N_INPUTS];
+    const secp256k1_xonly_pubkey *tx_input_ptrs[N_INPUTS];
+    secp256k1_xonly_pubkey tx_outputs[N_OUTPUTS];
+    secp256k1_xonly_pubkey *tx_output_ptrs[N_OUTPUTS];
+    int ret;
+    size_t i;
+
+    /* Before we can call actual API functions, we need to create a "context" */
+    secp256k1_context* ctx = secp256k1_context_create(SECP256K1_CONTEXT_NONE);
+    if (!fill_random(randomize, sizeof(randomize))) {
+        printf("Failed to generate randomness\n");
+        return 1;
```

Comment:
nit: could update the `main` return values in the example to use `EXIT_FAILURE`/`EXIT_SUCCESS` (done for other examples in #1654)